### ISSUE TITLE
panic during index rebuild with azure backend

### ISSUE THREAD

**Issue by jcollie** [restic/restic#1386]
## Output of `restic version`
```
# restic version
restic 0.7.3
compiled with go1.9 on linux/amd64
```
## How did you run restic exactly?

restic rebuild-index

## What backend/server/service did you use?

Azure

## Expected behavior

Rebuild the index

## Actual behavior
```
# restic rebuild-index
counting files in repo
[1:26:41] 100.00%  197020 / 197020 packs
finding old index files
storage: service returned error: StatusCode=413, ErrorCode=RequestBodyTooLarge, ErrorMessage=The request body is too large and exceeds the maximum permissible limit.
RequestId:059980c3-001e-0066-36ba-4d2ff1000000
Time:2017-10-25T17:56:17.7914151Z, RequestInitiated=Wed, 25 Oct 2017 17:56:17 GMT, RequestId=059980c3-001e-0066-36ba-4d2ff1000000, API Version=2016-05-31, QueryParameterName=, QueryParameterValue=
CreateBlockBlobFromReader
github.com/restic/restic/internal/backend/azure.(*Backend).Save
	src/github.com/restic/restic/internal/backend/azure/azure.go:166
github.com/restic/restic/internal/repository.(*Repository).SaveUnpacked
	src/github.com/restic/restic/internal/repository/repository.go:230
github.com/restic/restic/internal/repository.(*Repository).SaveJSONUnpacked
	src/github.com/restic/restic/internal/repository/repository.go:215
github.com/restic/restic/internal/index.Save
	src/github.com/restic/restic/internal/index/index.go:292
github.com/restic/restic/internal/index.(*Index).Save
	src/github.com/restic/restic/internal/index/index.go:263
main.rebuildIndex
	src/github.com/restic/restic/cmd/restic/cmd_rebuild_index.go:67
main.runRebuildIndex
	src/github.com/restic/restic/cmd/restic/cmd_rebuild_index.go:43
main.glob..func17
	src/github.com/restic/restic/cmd/restic/cmd_rebuild_index.go:21
github.com/restic/restic/vendor/github.com/spf13/cobra.(*Command).execute
	src/github.com/restic/restic/vendor/github.com/spf13/cobra/command.go:650
github.com/restic/restic/vendor/github.com/spf13/cobra.(*Command).ExecuteC
	src/github.com/restic/restic/vendor/github.com/spf13/cobra/command.go:729
github.com/restic/restic/vendor/github.com/spf13/cobra.(*Command).Execute
	src/github.com/restic/restic/vendor/github.com/spf13/cobra/command.go:688
main.main
	src/github.com/restic/restic/cmd/restic/main.go:69
runtime.main
	/usr/local/go/src/runtime/proc.go:185
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:2337
```

## Steps to reproduce the behavior

Reproduction might be difficult as it appears my repo may be corrupt.

## Do you have any idea what may have caused this?

Not exactly. I had to kill restic several times during a backup so that may have corrupted the repository.

## Do you have an idea how to solve the issue?

No.

**Comment by fd0** [comment]
Thanks for reporting this issue! It looks like the Azure API rejects uploads larger than 256 MiB: https://godoc.org/github.com/Azure/azure-sdk-for-go/storage#Blob.CreateBlockBlobFromReader

We need to handle this differently, but I'm not sure how.

@tamalsaha would you mind having a look?

**Comment by fd0** [comment]
Ping @diptadas

**Comment by fd0** [comment]
This is resolved in #1412, we're restricting index files to 100MiB. I'm closing this issue for now, please feel free to add further comments. Thanks!
